csrc/C++20/ranges

```diff
@@ -0,0 +1,81 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <type_traits>
+
+// This is backport of C++20 ranges library to C++17.
```

Comment:
Say in the comment which file you copied this from, for posterity.

Reply:
It is a class that I write myself to emulate the behavior of C++20's ranges. It is not copied from anywhere.